Build the book inside docker

bhesmans opened the pull request:
Followed the instruction in BUILD.md, but missed the theme (sphinx-book-theme\n\) and use the original tkz because the other does nt build

This is a simple version that only build the html version, could build the rest in the same way

bhesmans committed:
Build the html inside docker

Followed the instruction in BUILD.md, but missed the theme
(sphinx-book-theme\n\) and use the original tkz because the other does
nt build

Signed-off-by: Benjamin Hesmans <ben.hesmans@gmail.com>

obonaventure reviewed (changes requested):
Thanks. It works for me on my Mac as well. Would be nice to add support for spell checker - sphinx-build --keep-going -b spelling and also latexpdf and ePub to ensure that everything builds correctly. This will help contributors !

obonaventure commented:
docker gives one warning:

FROM ubuntu:24.04 as build

FromAsCasing: 'as' and 'FROM' keywords' casing do not match

bhesmans committed:
Dockerfile: add spelling check

The build will stop if there is any spelling mistake.

To be fixed:

principles/reliability.rst:838: : Spell check: Utilisation: Utilisation of the sliding window with modulo arithmetic.
protocols/email.rst:303: : Spell check: fb: 0x14fb.
protocols/ethernet.rst:56: : Spell check: utilisation: Impact of the frame length on the maximum channel utilisation .
protocols/ethernet.rst:56: : Spell check: utilisation: Impact of the frame length on the maximum channel utilisation .
protocols/wifi.rst:42: : Spell check: adhoc: An 802.11 independent or adhoc network.
protocols/routing.rst:14: : Spell check: Organisation: Organisation of a small Internet.
protocols/tcp.rst:400: : Spell check: Disambiguating: Disambiguating round-trip-time measurements with the .
protocols/wifi.rst:42: : Spell check: adhoc: An 802.11 independent or adhoc network.

Signed-off-by: Benjamin Hesmans <ben.hesmans@gmail.com>

bhesmans committed:
Dockerfile: only check for spelling mistake

Currently ignore the other warnings

Signed-off-by: Benjamin Hesmans <ben.hesmans@gmail.com>

bhesmans committed:
fix: spelling issues

Signed-off-by: Benjamin Hesmans <ben.hesmans@gmail.com>

bhesmans committed:
Docker: make epub

Signed-off-by: Benjamin Hesmans <ben.hesmans@gmail.com>

bhesmans committed:
docker: add pdf

Signed-off-by: Benjamin Hesmans <ben.hesmans@gmail.com>

bhesmans commented:
- docker warning is corrected
- spell check is done (+ a few fixes for that)
- pdf and epub also produced
